### PR TITLE
Add the Flaky Check

### PR DESCRIPTION
**Pull request by akuzm**
It runs the changed tests many times to find flakiness.

### CODE REVIEW

File: .github/gh_matrix_builder.py [hunk 1]
```diff
@@ -117,16 +119,18 @@ def build_apache_config(overrides):
...
-            "tsdb_build_args": "-DASSERTIONS=ON -DREQUIRE_ALL_TESTS=ON -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl",
+            "clang": "clang",
             "coverage": False,
-            "installcheck_args": "IGNORES='bgw_db_scheduler bgw_db_scheduler_fixed bgw_launcher pg_dump remote_connection compressed_collation'",
+            "cxx": "clang++",
             "extra_packages": "",
+            "installcheck_args": "IGNORES='bgw_db_scheduler bgw_db_scheduler_fixed bgw_launcher pg_dump remote_connection compressed_collation'",
+            "llvm_config": "/usr/local/opt/llvm/bin/llvm-config",
+            "os": "macos-11",
+            "pg": PG12_LATEST,
```
Comment (nikkhils) [inline .github/gh_matrix_builder.py:130]: Should we still use PG12 here?

**Review by nikkhils — APPROVED** [review]
LGTM overall.

File: .github/gh_matrix_builder.py (hunk 1 above)
Comment (akuzm) [reply .github/gh_matrix_builder.py:130]: I don't know... probably should remove support when we start supporting PG 15.

Comment (fabriziomello) [reply .github/gh_matrix_builder.py:130]: The plan is remove it when we have full PG15 support!

File: test/sql/agg_bookends.sql.in [hunk 2]
```diff
@@ -31,3 +31,5 @@ SET timescaledb.enable_optimizations TO true;
 \o
 
 :DIFF_CMD
+
+
```
Comment (fabriziomello) [inline test/sql/agg_bookends.sql.in:35]:
```suggestion
```
Leftover?

**Review by fabriziomello — APPROVED** [review]

File: test/sql/agg_bookends.sql.in (hunk 2 above)
Comment (akuzm) [reply test/sql/agg_bookends.sql.in:35]: Yep, I used them to test that the check works.

**Pull request merged**